cmake/init.cmake

```diff
@@ -40,7 +40,7 @@ if(NOT DEFINED DDS_ROOT)
     set(OPENDDS_LIB_DIR "${DDS_ROOT}/lib")
   endif()
 
-else()
+elseif(NOT DEFINED DDS_ALLOW_ENV_REDEFINES)
```

Comment:
This should start with `OPENDDS` and should probably this should probably be an [option/cache variable](https://github.com/objectcomputing/OpenDDS/blob/26d4b1dfc663c1347169a4ca3082f96a04c309d9/cmake/init.cmake#L11-L12). You could also make the include filename only thing a global setting if you want.

Can you also document this in and everything else [`cmake.md`](https://github.com/objectcomputing/OpenDDS/blob/master/docs/cmake.md#cache-variablesoptions-understood-by-opendds)?

Reply:
I accidentally clicked approved and I can't seem to change it.